src/data_structures/suffix_trees.rs

```diff
@@ -7,7 +7,7 @@ const ROOT: usize = 1;
 #[derive(Debug)]
 enum End {
     Root,
-    End,
+    EOS,
```

Comment:
Good name change.  I would have perhapsed changed enum End to enum End_state. but that's just me.

Reply:
@Gavin-Bowers and @tw-edu will be making more changes concurrently touching this enum, so it will be a separate task to convince them from this PR.